app/controllers/foreman_bootdisk/api/v2/disks_controller.rb

```diff
@@ -37,8 +37,13 @@ def host
           host = @disk
           if params[:full]
             return unless bootdisk_type_allowed?('full_host')
-            ForemanBootdisk::ISOGenerator.generate_full_host(host) do |iso|
-              send_file(iso, filename: "#{host.name}#{ForemanBootdisk::ISOGenerator.token_expiry(host)}.iso")
+            begin
+              ForemanBootdisk::ISOGenerator.generate_full_host(host) do |iso|
+                send_file(iso, filename: "#{host.name}#{ForemanBootdisk::ISOGenerator.token_expiry(host)}.iso")
+              end
+            rescue ::Foreman::Exception => e
+              raise e unless e.code == 'ERF42-2893'
+              render json: { error: _('Host is not in build mode')}, status: :method_not_allowed
```

Comment:
Please use `render_error` from the base controller, this does not create a valid JSON and CLI returns just:

```
[lzap@nuc cli]$ hammer bootdisk host --host allan-ocampo.dummy.lan --full true --file /dev/null --force
Failed to download host disk image:
  Error: 405 Method Not Allowed
```

With debug I can see JSON is returned but CLI ignores it, there is I think root element missing or something.

```
[ INFO 2021-09-03T10:05:10 API] GET /bootdisk/api/hosts/16
[DEBUG 2021-09-03T10:05:10 API] Params: {
    "full" => true
}
[DEBUG 2021-09-03T10:05:10 API] Headers: {
    :params => {
        "full" => true
    }
}
[DEBUG 2021-09-03T10:05:10 API] Using authenticator: HammerCLIForeman::Api::SessionAuthenticatorWrapper
[ERROR 2021-09-03T10:05:11 API] 405 Method Not Allowed
[DEBUG 2021-09-03T10:05:11 API] {
    "error" => "Host is not in build mode"
}
[DEBUG 2021-09-03T10:05:11 Exception] Using exception handler HammerCLIForeman::ExceptionHandler#handle_general_exception
[ERROR 2021-09-03T10:05:11 Exception] Error: 405 Method Not Allowed
Failed to download host disk image:
  Error: 405 Method Not Allowed
```

Reply:
Call has been updated